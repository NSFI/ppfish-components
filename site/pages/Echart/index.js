import Markdown from '../../../libs/markdown';

export default class Echart extends Markdown {
  document(locale) {
    return require(`../../docs/${locale}/loading.md`);
  }
}
