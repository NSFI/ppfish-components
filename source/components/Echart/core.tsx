import * as React from 'react';
import classNames from 'classnames';
import ReactEchartsCore from 'echarts-for-react/lib/core';
import defaultTheme from './default-theme';

interface EventMap {
  [key: string]: (...args: any[]) => any;
}

export interface EchartCoreRef {
  getInstance: () => any;
  echarts_react: any;
}

type Option = { [key: string]: any };
export interface EchartCoreProps {
  prefixCls?: string;
  option?: Option;
  opts?: Option;
  events?: EventMap;
  className?: string;
  style?: React.CSSProperties;
  theme?: string;
  notMerge?: boolean;
  lazyUpdate?: boolean;
  onChartReady?: () => void;
  loadingOption?: Option;
  showLoading?: boolean;
  echarts?: any;
}

const InternalEchartCore: React.ForwardRefRenderFunction<EchartCoreRef, EchartCoreProps> = (
  props,
  ref,
) => {
  const echartCoreRef = React.useRef<any>();

  React.useImperativeHandle(ref, () => ({
    getInstance: () => {
      return echartCoreRef.current;
    },
    echarts_react: echartCoreRef.current,
  }));

  const {
    prefixCls,
    className,
    style,
    theme,
    option,
    opts,
    events,
    notMerge,
    lazyUpdate,
    onChartReady,
    loadingOption,
    showLoading,
    echarts,
  } = props;
  return (
    <ReactEchartsCore
      ref={e => (echartCoreRef.current = e)}
      echarts={echarts}
      className={classNames(prefixCls, className)}
      style={style}
      option={option}
      opts={opts}
      onEvents={events}
      theme={theme}
      notMerge={notMerge}
      lazyUpdate={lazyUpdate}
      onChartReady={onChartReady}
      loadingOption={loadingOption}
      showLoading={showLoading}
    />
  );
};

const EchartCore = React.forwardRef<EchartCoreRef, EchartCoreProps>(InternalEchartCore);

EchartCore.displayName = 'EchartCore';

EchartCore.defaultProps = {
  prefixCls: 'fishd-chart',
  theme: defaultTheme,
  echarts: {},
  notMerge: false,
  lazyUpdate: false,
  style: { height: '300px' },
  className: '',
  onChartReady: () => {},
  showLoading: false,
  loadingOption: null,
  opts: {},
};

export default EchartCore;
