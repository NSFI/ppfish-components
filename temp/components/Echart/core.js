var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import ReactEchartsCore from 'echarts-for-react/lib/core';
import defaultTheme from './default-theme';
var EchartCore = /** @class */ (function (_super) {
    __extends(EchartCore, _super);
    function EchartCore(props) {
        var _this = _super.call(this, props) || this;
        _this.echarts_react = null;
        return _this;
    }
    EchartCore.prototype.getInstance = function () {
        // @ts-ignore
        return this.echarts_react.getEchartsInstance();
    };
    EchartCore.prototype.render = function () {
        var _this = this;
        var _a = this.props, prefixCls = _a.prefixCls, className = _a.className, style = _a.style, theme = _a.theme, option = _a.option, opts = _a.opts, events = _a.events, notMerge = _a.notMerge, lazyUpdate = _a.lazyUpdate, onChartReady = _a.onChartReady, loadingOption = _a.loadingOption, showLoading = _a.showLoading, echarts = _a.echarts;
        return (React.createElement(ReactEchartsCore, { ref: function (e) { return (_this.echarts_react = e); }, echarts: echarts, className: classNames(prefixCls, className), style: style, option: option, opts: opts, onEvents: events, theme: theme, notMerge: notMerge, lazyUpdate: lazyUpdate, onChartReady: onChartReady, loadingOption: loadingOption, showLoading: showLoading }));
    };
    EchartCore.propTypes = {
        prefixCls: PropTypes.string,
        option: PropTypes.object,
        opts: PropTypes.object,
        events: PropTypes.object,
        className: PropTypes.string,
        style: PropTypes.object,
        theme: PropTypes.string,
        notMerge: PropTypes.bool,
        lazyUpdate: PropTypes.bool,
        onChartReady: PropTypes.func,
        loadingOption: PropTypes.object,
        showLoading: PropTypes.bool,
        echarts: PropTypes.object
    };
    EchartCore.defaultProps = {
        prefixCls: 'fishd-chart',
        theme: defaultTheme,
        echarts: {},
        notMerge: false,
        lazyUpdate: false,
        style: { height: '300px' },
        className: '',
        onChartReady: function () { },
        showLoading: false,
        loadingOption: null,
        onEvents: {},
        opts: {}
    };
    return EchartCore;
}(Component));
export default EchartCore;
