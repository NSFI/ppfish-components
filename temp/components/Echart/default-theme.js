import echarts from 'echarts/lib/echarts';
var themeName = 'fishd-default-theme';
var theme = {
    color: ['#4d6aff', '#26bd71', '#ffaf0f', '#f8355c', '#7231f5'],
    backgroundColor: '#ffffff',
    textStyle: {},
    title: {
        textStyle: {
            color: '#333333',
        },
        subtextStyle: {
            color: '#aaaaaa',
        },
    },
    line: {
        itemStyle: {
            normal: {
                borderWidth: '3',
            },
        },
        lineStyle: {
            normal: {
                width: '3',
            },
        },
        symbolSize: '7',
        symbol: 'emptyCircle',
        smooth: false,
    },
    radar: {
        itemStyle: {
            normal: {
                borderWidth: '3',
            },
        },
        lineStyle: {
            normal: {
                width: '3',
            },
        },
        symbolSize: '7',
        symbol: 'emptyCircle',
        smooth: false,
    },
    bar: {
        itemStyle: {
            normal: {
                barBorderWidth: '0',
                barBorderColor: '#cccccc',
            },
            emphasis: {
                barBorderWidth: '0',
                barBorderColor: '#cccccc',
            },
        },
    },
    pie: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    scatter: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    boxplot: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    parallel: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    sankey: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    funnel: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    gauge: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
            emphasis: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
    },
    candlestick: {
        itemStyle: {
            normal: {
                color: '#e01f54',
                color0: '#001852',
                borderColor: '#f5e8c8',
                borderColor0: '#b8d2c7',
                borderWidth: 1,
            },
        },
    },
    graph: {
        itemStyle: {
            normal: {
                borderWidth: '0',
                borderColor: '#cccccc',
            },
        },
        lineStyle: {
            normal: {
                width: 1,
                color: '#aaaaaa',
            },
        },
        symbolSize: '7',
        symbol: 'emptyCircle',
        smooth: false,
        color: ['#4d6aff', '#26bd71', '#ffaf0f', '#f8355c', '#7231f5'],
        label: {
            normal: {
                textStyle: {
                    color: '#eeeeee',
                },
            },
        },
    },
    map: {
        itemStyle: {
            normal: {
                areaColor: '#eeeeee',
                borderColor: '#444444',
                borderWidth: 0.5,
            },
            emphasis: {
                areaColor: 'rgba(255,215,0,0.8)',
                borderColor: '#444444',
                borderWidth: 1,
            },
        },
        label: {
            normal: {
                textStyle: {
                    color: '#000000',
                },
            },
            emphasis: {
                textStyle: {
                    color: 'rgb(100,0,0)',
                },
            },
        },
    },
    geo: {
        itemStyle: {
            normal: {
                areaColor: '#eeeeee',
                borderColor: '#444444',
                borderWidth: 0.5,
            },
            emphasis: {
                areaColor: 'rgba(255,215,0,0.8)',
                borderColor: '#444444',
                borderWidth: 1,
            },
        },
        label: {
            normal: {
                textStyle: {
                    color: '#000000',
                },
            },
            emphasis: {
                textStyle: {
                    color: 'rgb(100,0,0)',
                },
            },
        },
    },
    categoryAxis: {
        axisLine: {
            show: true,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisTick: {
            show: false,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisLabel: {
            show: true,
            textStyle: {
                color: '#666666',
            },
        },
        splitLine: {
            show: true,
            lineStyle: {
                color: ['#f0f3f7'],
            },
        },
        splitArea: {
            show: false,
            areaStyle: {
                color: ['rgba(250,250,250,0.3)', 'rgba(200,200,200,0.3)'],
            },
        },
    },
    valueAxis: {
        axisLine: {
            show: true,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisTick: {
            show: false,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisLabel: {
            show: true,
            textStyle: {
                color: '#666666',
            },
        },
        splitLine: {
            show: true,
            lineStyle: {
                color: ['#f0f3f7'],
            },
        },
        splitArea: {
            show: false,
            areaStyle: {
                color: ['rgba(250,250,250,0.3)', 'rgba(200,200,200,0.3)'],
            },
        },
    },
    logAxis: {
        axisLine: {
            show: true,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisTick: {
            show: false,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisLabel: {
            show: true,
            textStyle: {
                color: '#666666',
            },
        },
        splitLine: {
            show: true,
            lineStyle: {
                color: ['#f0f3f7'],
            },
        },
        splitArea: {
            show: false,
            areaStyle: {
                color: ['rgba(250,250,250,0.3)', 'rgba(200,200,200,0.3)'],
            },
        },
    },
    timeAxis: {
        axisLine: {
            show: true,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisTick: {
            show: false,
            lineStyle: {
                color: '#f0f3f7',
            },
        },
        axisLabel: {
            show: true,
            textStyle: {
                color: '#666666',
            },
        },
        splitLine: {
            show: true,
            lineStyle: {
                color: ['#f0f3f7'],
            },
        },
        splitArea: {
            show: false,
            areaStyle: {
                color: ['rgba(250,250,250,0.3)', 'rgba(200,200,200,0.3)'],
            },
        },
    },
    toolbox: {
        iconStyle: {
            normal: {
                borderColor: '#999999',
            },
            emphasis: {
                borderColor: '#666666',
            },
        },
    },
    legend: {
        textStyle: {
            color: '#666666',
        },
    },
    tooltip: {
        axisPointer: {
            lineStyle: {
                color: '#cccccc',
                width: 1,
            },
            crossStyle: {
                color: '#cccccc',
                width: 1,
            },
        },
    },
    timeline: {
        lineStyle: {
            color: '#293c55',
            width: 1,
        },
        itemStyle: {
            normal: {
                color: '#293c55',
                borderWidth: 1,
            },
            emphasis: {
                color: '#a9334c',
            },
        },
        controlStyle: {
            normal: {
                color: '#293c55',
                borderColor: '#293c55',
                borderWidth: 0.5,
            },
            emphasis: {
                color: '#293c55',
                borderColor: '#293c55',
                borderWidth: 0.5,
            },
        },
        checkpointStyle: {
            color: '#e43c59',
            borderColor: 'rgba(194,53,49,0.5)',
        },
        label: {
            normal: {
                textStyle: {
                    color: '#293c55',
                },
            },
            emphasis: {
                textStyle: {
                    color: '#293c55',
                },
            },
        },
    },
    visualMap: {
        color: ['#fb4e0d', '#ffbc00'],
    },
    dataZoom: {
        backgroundColor: 'rgba(47,69,84,0)',
        dataBackgroundColor: 'rgba(47,69,84,0.3)',
        fillerColor: 'rgba(167,183,204,0.4)',
        handleColor: '#a7b7cc',
        handleSize: '100%',
        textStyle: {
            color: '#333333',
        },
    },
    markPoint: {
        label: {
            normal: {
                textStyle: {
                    color: '#eeeeee',
                },
            },
            emphasis: {
                textStyle: {
                    color: '#eeeeee',
                },
            },
        },
    },
};
echarts.registerTheme(themeName, theme);
export default themeName;
