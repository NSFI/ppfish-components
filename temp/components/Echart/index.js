var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import ReactEcharts from 'echarts-for-react';
import defaultTheme from './default-theme';
var Echart = /** @class */ (function (_super) {
    __extends(Echart, _super);
    function Echart(props) {
        var _this = _super.call(this, props) || this;
        _this.echarts_react = null;
        return _this;
    }
    Echart.prototype.getInstance = function () {
        // @ts-ignore
        return this.echarts_react.getEchartsInstance();
    };
    Echart.prototype.render = function () {
        var _this = this;
        var _a = this.props, prefixCls = _a.prefixCls, className = _a.className, style = _a.style, theme = _a.theme, option = _a.option, opts = _a.opts, events = _a.events, notMerge = _a.notMerge, lazyUpdate = _a.lazyUpdate, onChartReady = _a.onChartReady, loadingOption = _a.loadingOption, showLoading = _a.showLoading;
        return (React.createElement(ReactEcharts, { ref: function (e) { return (_this.echarts_react = e); }, className: classNames(prefixCls, className), style: style, option: option, opts: opts, onEvents: events, theme: theme, notMerge: notMerge, lazyUpdate: lazyUpdate, onChartReady: onChartReady, loadingOption: loadingOption, showLoading: showLoading }));
    };
    Echart.propTypes = {
        prefixCls: PropTypes.string,
        option: PropTypes.object,
        opts: PropTypes.object,
        events: PropTypes.object,
        className: PropTypes.string,
        style: PropTypes.object,
        theme: PropTypes.string,
        notMerge: PropTypes.bool,
        lazyUpdate: PropTypes.bool,
        onChartReady: PropTypes.func,
        loadingOption: PropTypes.object,
        showLoading: PropTypes.bool
    };
    Echart.defaultProps = {
        prefixCls: 'fishd-chart',
        theme: defaultTheme,
        notMerge: false,
        lazyUpdate: false,
        style: { height: '300px' },
        className: '',
        onChartReady: function () { },
        showLoading: false,
        loadingOption: null,
        onEvents: {},
        opts: {}
    };
    return Echart;
}(Component));
export default Echart;
