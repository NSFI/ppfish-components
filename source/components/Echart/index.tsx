import * as React from 'react';
import classNames from 'classnames';
import ReactEcharts from 'echarts-for-react';
import defaultTheme from './default-theme';
import { EchartCoreProps } from './core';

type EchartProps = Omit<EchartCoreProps, 'echarts'>;

export interface EchartRef {
  getInstance: () => any;
  echarts_react: any;
}

const InernalEchart: React.ForwardRefRenderFunction<EchartRef, EchartProps> = (props, ref) => {
  const {
    prefixCls,
    className,
    style,
    theme,
    option,
    opts,
    events,
    notMerge,
    lazyUpdate,
    onChartReady,
    loadingOption,
    showLoading,
  } = props;

  const echartRef = React.useRef<any>();

   React.useImperativeHandle(ref, () => ({
    getInstance: () => {
      return echartRef.current;
    },
    echarts_react: echartRef.current
  }));

  return (
    <ReactEcharts
      ref={e => {echartRef.current = e}}
      className={classNames(prefixCls, className)}
      style={style}
      option={option}
      opts={opts}
      onEvents={events}
      theme={theme}
      notMerge={notMerge}
      lazyUpdate={lazyUpdate}
      onChartReady={onChartReady}
      loadingOption={loadingOption}
      showLoading={showLoading}
    />
  );
};

const Echart = React.forwardRef<EchartRef, EchartProps>(InernalEchart);

Echart.defaultProps = {
  prefixCls: 'fishd-chart',
  theme: defaultTheme,
  notMerge: false,
  lazyUpdate: false,
  style: { height: '300px' },
  className: '',
  onChartReady: () => {},
  showLoading: false,
  loadingOption: null,
  opts: {},
};

Echart.displayName = 'Echart';

export default Echart;
