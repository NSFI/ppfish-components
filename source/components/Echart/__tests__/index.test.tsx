import React from 'react';
import { mount } from 'enzyme';
import Echart from '../index';

describe('Echart', () => {
  const option = {
    tooltip: {
      trigger: 'item',
      formatter: '{a} <br/>{b}: {c} ({d}%)',
    },
    legend: {
      orient: 'vertical',
      x: 'left',
      data: ['直接访问', '邮件营销', '联盟广告', '视频广告', '搜索引擎'],
    },
    series: [
      {
        name: '访问来源',
        type: 'pie',
        radius: ['50%', '70%'],
        avoidLabelOverlap: false,
        label: {
          normal: {
            show: false,
            position: 'center',
          },
          emphasis: {
            show: true,
            textStyle: {
              fontSize: '30',
              fontWeight: 'bold',
            },
          },
        },
        labelLine: {
          normal: {
            show: false,
          },
        },
        data: [
          { value: 335, name: '直接访问' },
          { value: 310, name: '邮件营销' },
          { value: 234, name: '联盟广告' },
          { value: 135, name: '视频广告' },
          { value: 1548, name: '搜索引擎' },
        ],
      },
    ],
  };

  it('should give correct ref', () => {
    let ref = null;
    const wrapper = mount(
      <Echart
        className="my-echarts"
        ref={e => {
          ref = e;
        }}
        option={option}
        style={{ width: 500, height: 500 }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(ref).toHaveProperty('getInstance');
    expect(ref).toHaveProperty('echarts_react');
    expect(ref.getInstance()).toEqual(ref.echarts_react);
  });
});
