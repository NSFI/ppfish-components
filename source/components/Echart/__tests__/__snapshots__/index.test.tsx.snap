// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Echart should give correct ref 1`] = `
<Echart
  className="my-echarts"
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  option={
    Object {
      "legend": Object {
        "data": Array [
          "直接访问",
          "邮件营销",
          "联盟广告",
          "视频广告",
          "搜索引擎",
        ],
        "orient": "vertical",
        "x": "left",
      },
      "series": Array [
        Object {
          "avoidLabelOverlap": false,
          "data": Array [
            Object {
              "name": "直接访问",
              "value": 335,
            },
            Object {
              "name": "邮件营销",
              "value": 310,
            },
            Object {
              "name": "联盟广告",
              "value": 234,
            },
            Object {
              "name": "视频广告",
              "value": 135,
            },
            Object {
              "name": "搜索引擎",
              "value": 1548,
            },
          ],
          "label": Object {
            "emphasis": Object {
              "show": true,
              "textStyle": Object {
                "fontSize": "30",
                "fontWeight": "bold",
              },
            },
            "normal": Object {
              "position": "center",
              "show": false,
            },
          },
          "labelLine": Object {
            "normal": Object {
              "show": false,
            },
          },
          "name": "访问来源",
          "radius": Array [
            "50%",
            "70%",
          ],
          "type": "pie",
        },
      ],
      "tooltip": Object {
        "formatter": "{a} <br/>{b}: {c} ({d}%)",
        "trigger": "item",
      },
    }
  }
  opts={Object {}}
  prefixCls="fishd-chart"
  showLoading={false}
  style={
    Object {
      "height": 500,
      "width": 500,
    }
  }
  theme="fishd-default-theme"
>
  <EchartsReact
    className="fishd-chart my-echarts"
    echarts={Object {}}
    lazyUpdate={false}
    loadingOption={null}
    notMerge={false}
    onChartReady={[Function]}
    onEvents={Object {}}
    option={
      Object {
        "legend": Object {
          "data": Array [
            "直接访问",
            "邮件营销",
            "联盟广告",
            "视频广告",
            "搜索引擎",
          ],
          "orient": "vertical",
          "x": "left",
        },
        "series": Array [
          Object {
            "avoidLabelOverlap": false,
            "data": Array [
              Object {
                "name": "直接访问",
                "value": 335,
              },
              Object {
                "name": "邮件营销",
                "value": 310,
              },
              Object {
                "name": "联盟广告",
                "value": 234,
              },
              Object {
                "name": "视频广告",
                "value": 135,
              },
              Object {
                "name": "搜索引擎",
                "value": 1548,
              },
            ],
            "label": Object {
              "emphasis": Object {
                "show": true,
                "textStyle": Object {
                  "fontSize": "30",
                  "fontWeight": "bold",
                },
              },
              "normal": Object {
                "position": "center",
                "show": false,
              },
            },
            "labelLine": Object {
              "normal": Object {
                "show": false,
              },
            },
            "name": "访问来源",
            "radius": Array [
              "50%",
              "70%",
            ],
            "type": "pie",
          },
        ],
        "tooltip": Object {
          "formatter": "{a} <br/>{b}: {c} ({d}%)",
          "trigger": "item",
        },
      }
    }
    opts={Object {}}
    shouldSetOption={[Function]}
    showLoading={false}
    style={
      Object {
        "height": 500,
        "width": 500,
      }
    }
    theme="fishd-default-theme"
  >
    <div
      className="echarts-for-react fishd-chart my-echarts"
      style={
        Object {
          "height": 500,
          "width": 500,
        }
      }
    />
  </EchartsReact>
</Echart>
`;
